### ISSUE TITLE
Add full %s support

### ISSUE THREAD

**Issue by rokath** [rokath/trice#21]
Legacy projects can contain statements like `printf( "In year %d town %9s had %4d sunny hours\n", 2019, "Berlin", 1925); `. With TRICE this should be possible:  Convert above to  `trice( "In year %d town %9s had %4d sunny hours\n", 2019, "Berlin", 1925);` and write a function `trice(...)` which internally parses the format string in a loop like that:
```
int trice (...){
Loop until end of formatstring:
    triceString( "...till next%"
    switch % {
    case %d: TRICE32( "%d", value)
    case %03d: TRICE32( "%03d", value) 
    case %s: triceString( pString)
    default: error // extend case for needed formats
}}
```
Alternatively a source code parsing could automatically translate the printf lines into lines like  `TRICE32_1( Id(1234), "In year %d town ", 2019 ); TriceString( 9, "Berlin"); TRICE32_1( Id(567), " had %4 sunny hours\n", 1925 );` This is a more universal and faster approach but spoils a bit the legacy code and takes more memory.  The user should be able to choose.

**Comment by rokath** [comment]
The `trice()`function should handle specifiers `%03d` as with `03` parametrized `%d` so it can get implemented in an universal way.

**Comment by rokath** [comment]
implemented tricePrintfAdapter() in release 0.4.2 solves this
